client/src/cmdhfst.c

```diff
@@ -473,57 +452,72 @@ static int cmd_hf_st_ndef(const char *Cmd) {
-            case '0':
-                state[0] = 0x26;  //Disable protection
-                cmdp++;
-                break;
-            case '1':
-                state[0] = 0x28;  //Enable protection
-                cmdp++;
-                break;
-            case 'r':
-                state[2] = 0x01;
-                cmdp++;
-                break;
-            case 'w':
-                state[2] = 0x02;
-                cmdp++;
-                break;
-            case 'p':
-                param_gethex_ex(Cmd, cmdp + 1, pwd, &pwdlen);
-                pwdlen >>= 1;
-                cmdp += 2;
-                break;
-            default:
-                PrintAndLogEx(WARNING, "Unknown parameter '%c'", param_getchar(Cmd, cmdp));
-                errors = true;
-                break;
-        }
-    }
+    uint8_t state[3] = {0x26, 0, 0x02};
+
+    bool disable_protection = false;
+    bool enable_protection = false;
+    bool read_protection = false;
+    bool write_protection = false;
+
+    CLIParserContext *ctx;
+    CLIParserInit(&ctx, "hf st protect",
+                  "Change read or write protection for NFC Data Exchange Format (NDEF) file on ST25TA",
+                  "hf st protect -p 82E80053D4CA5C0B656D852CC696C8A1 -r -e -> enable read protection\n"
+                  "hf st protect -p 82E80053D4CA5C0B656D852CC696C8A1 -w -d -> disable write protection\n");
+
+    void *argtable[] = {
+        arg_param_begin,
+        arg_lit0("e",  "enable",            "enable protection"),
+        arg_lit0("d",  "disable",           "disable protection (default)"),        
+        arg_lit0("r",  "read",              "change read protection"),
+        arg_lit0("w",  "write",             "change write protection (default)"),
+        arg_str1("p", "password", "<hex>", "16 byte write password"),
+        arg_param_end
+    };
+    CLIExecWithReturn(ctx, Cmd, argtable, false);
+
+    enable_protection = arg_get_lit(ctx, 1);
+    disable_protection = arg_get_lit(ctx, 2);
+    read_protection = arg_get_lit(ctx, 3);
+    write_protection = arg_get_lit(ctx, 4);
+
+    CLIGetHexWithReturn(ctx, 5, pwd, &pwdlen);
+
+    CLIParserFree(ctx);        
 
     //Validations
 
-    if (state[2] == 0x00) {
-        PrintAndLogEx(WARNING, "Missing action (r)ead or (w)rite");
-        errors = true;
+    if (enable_protection && disable_protection) {
+        PrintAndLogEx(ERR, "Must specify either enable or disable protection, not both");
+        return PM3_EINVARG;
+    } else {
+        if (enable_protection) {
+            state[0] = 0x28;
+        }
+        if (disable_protection) {
+            state[0] = 0x26;
+        }
     }
-    if (pwdlen != 16) {
-        PrintAndLogEx(WARNING, "Missing 16 byte password");
-        errors = true;
+
+    if (read_protection && write_protection) {
+        PrintAndLogEx(ERR, "Must specify either read or write protection, not both");
+        return PM3_EINVARG;
+    } else {
```

Comment:
No need for `else` if you return

Reply:
The else is if the error condition is not met.  I guess I could refactor it and do it differently if you would prefer.

Reply:
obviously but since you are _returning_ ,   there is no use to have a _else_  
So you can make less indentation,  less "nested if's"   making the code a bit easier to read.   
Not much of a refactor to talk about.